api.go

```diff
@@ -53,3 +59,15 @@ func WithSettingEngine(s SettingEngine) func(a *API) {
 		a.settingEngine = &s
 	}
 }
+
+func WithInterceptor(interceptor Interceptor) func(a *API) {
+	return func(a *API) {
+		a.interceptors = append(a.interceptors, interceptor)
+	}
+}
+
+func ClearInterceptors() func(a *API) {
```

Comment:
What do you think of copying `MediaEngine` design? You get a list of well configured defaults OR you can customize them all.

`RegisterDefaultInterceptors` or w/e we call it will be easy to read so people can understand what is happening!

Reply:
Added a similar usage example to save-to-disk example.